### PR TITLE
tag combinator support char type

### LINKED ISSUE #407: the trait SliceLen is not implemented for char
### Please complete the following tasks

- [X] I have searched the [discussions](https://github.com/winnow-rs/winnow/discussions)
- [X] I have searched the [open](https://github.com/winnow-rs/winnow/issues) and [rejected](https://github.com/winnow-rs/winnow/issues?q=is%3Aissue+label%3AS-wont-fix+is%3Aclosed) issues

### rust version

rustc 1.74.0 (79e9716c9 2023-11-13)

### winnow version

0.5.31

### Minimal reproducible code

```rust
use std::num::{NonZeroU32, NonZeroU8};
use winnow::{Parser, PResult};
use winnow::ascii::space1;
use winnow::error::{ContextError, ParseError, ParserError};
use winnow::token::{tag, take};

fn main() {
    assert!(parse_human_format("2022-07-01 00:40:05 +08:00").is_ok());
    assert!(parse_human_format("2022-07-00 00:40:05 +08:00").is_err());
}


fn u8_len2(input: &mut &str) -> PResult<u8> {
    take(2_usize).try_map(str::parse).parse_next(input)
}

fn non_zero_u8_len2(input: &mut &str) -> PResult<NonZeroU8> {
    take(2_usize).try_map(str::parse).parse_next(input)
}

//
fn non_zero_u32_len4(input: &mut &str) -> PResult<NonZeroU32> {
    take(4_usize).try_map(str::parse).parse_next(input)
}

// 2022-07-14 00:40:05 +08:00
fn parse_human_format(input: &str) -> Result<(), ParseError<&str, ContextError>> {
    (non_zero_u32_len4,
     tag('-'),
     non_zero_u8_len2,
     tag('-'),
     non_zero_u8_len2,
     space1,
     u8_len2,
     tag(":"),
     u8_len2,
     tag(":"),
     u8_len2,
     space1,
     tag("+"),
     u8_len2,
     tag(":"),
     u8_len2,
    ).parse(input)?;
    Ok(())
}
```

* the  complier detail error
```diff
error[E0277]: the trait bound `char: SliceLen` is not satisfied
   --> src/main.rs:29:10
    |
29  |      tag('-'),
+    |      --- ^^^ the trait `SliceLen` is not implemented for `char`
    |      |
    |      required by a bound introduced by this call
    |
    = help: the following other types implement trait `SliceLen`:
              Located<I>
              Stateful<I, S>
              Partial<I>
              Caseless<S>
              [T; LEN]
              &'a [T]
              &'a [T; LEN]
              &'a str
            and 3 others
note: required by a bound in `winnow::token::tag`
   --> ~/.cargo/registry/src/winnow-0.5.30/src/token/mod.rs:142:8
    |
138 | pub fn tag<T, I, Error: ParserError<I>>(tag: T) -> impl Parser<I, <I as Stream>::Slice, Error>
    |        --- required by a bound in this function
...
142 |     T: SliceLen + Clone,
    |        ^^^^^^^^ required by this bound in `tag`


```

### Steps to reproduce the bug with the above code

exec above rust code,cause error

### Actual Behaviour

Maybe tag support `char`

### Expected Behaviour

-

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by baoyachi**
char convert to &[u8] ref: https://doc.rust-lang.org/stable/src/alloc/string.rs.html#2629-2634

 fix #407

### CODE REVIEW

File: src/stream/mod.rs [hunk 1]
```diff
@@ -1625,6 +1625,19 @@ impl<'a, 'b> Compare<&'b [u8]> for &'a [u8] {
     }
 }
 
+impl<'a> Compare<char> for &'a [u8] {
```
Comment (epage) [inline src/stream/mod.rs:1628]: If we go forward with this, we should cover all of the other  Stream types and `Caseless<char>`

**Comment by baoyachi** [comment]
For this example, both `one_of` and `char` are suitable. 
However, in certain scenarios, semantically,
`one_of` tends to imply a multiple choice selection where only one option is chosen.
Using the `char` type requires hardcoding, and it does not  passing parameters into a tag.

**Comment by epage** [comment]
Weird, no idea why that closed it.

**Comment by epage** [comment]
Recreated it in #417

**Pull request closed without merging**